asati/cli.py

```diff
@@ -20,7 +24,67 @@ def main(*, verbose: bool) -> None:
     logger.info(configure_sentry())
     logger.info("Running process")
 
-    # Do things here!
+    ssm_client = SSMClient()
+    last_accession_parameter = os.environ["LAST_ACCESSION_PARAMETER"]
+    last_accession_uri_id = ssm_client.get_parameter(last_accession_parameter)
+    logger.info(
+        f"SSM parameter '{last_accession_parameter}' retrieved: '{last_accession_uri_id}'"
+    )
+
+    asnake_client = ASnakeClient(
+        baseurl=os.environ["ARCHIVESSPACE_URL"],
+        username=os.environ["ARCHIVESSPACE_USER"],
+        password=os.environ["ARCHIVESSPACE_PASSWORD"],
+    )
+    accession_uri_ids = asnake_client.get(
+        "/repositories/2/accessions?all_ids=true"
+    ).json()
+
+    if max(accession_uri_ids) <= int(last_accession_uri_id):
+        logger.info("No new accessions to add to Airtable.")
+    else:
+        airtable_api = pyairtable.Api(os.environ["AIRTABLE_TOKEN"])
+        airtable_table = airtable_api.table(
+            os.environ["AIRTABLE_BASE_ID"], os.environ["AIRTABLE_TABLE_NAME"]
+        )
+        logger.info(
+            f"Airtable client configured for base '{airtable_table.base}'"
+            f" and table '{airtable_table.name}'"
+        )
+        accessions_to_add = sorted(
+            [
+                accession_uri_id
+                for accession_uri_id in accession_uri_ids
+                if accession_uri_id > int(last_accession_uri_id)
+            ]
+        )
+        logger.info(f"Adding the following accessions to Airtable: {accessions_to_add}")
```

Comment:
Thanks for adding!  I dig it in the output.

Reply:
Related to the "early exit" comment above, perhaps you could move this list creation _before_ the early exit, and then you could do something like:

```python
# determine new accessions to add
accessions_to_add = sorted(
    [
        accession_uri_id
        for accession_uri_id in accession_uri_ids
        if accession_uri_id > int(last_accession_uri_id)
    ]
)

# exit if no new accessions to add
if not accessions_to_add:
    logger.info("No new accessions to add to Airtable.")
    return
```

Reply:
And maybe even a bit _more_ reshuffling could reduce comments and let the logging statements speak for themselves:

```python
# determine accessions to add, exit early if none
accessions_to_add = sorted(
    [
        accession_uri_id
        for accession_uri_id in accession_uri_ids
        if accession_uri_id > int(last_accession_uri_id)
    ]
)
if not accessions_to_add:
    logger.info("No new accessions to add to Airtable.")
    return
logger.info(f"Adding the following accessions to Airtable: {accessions_to_add}")
```

More than just style, I think it requires only one place to check the SSM parameter value against the list of ASpace ids, which avoids errors creeping in over time if that needs adjustment.

Reply:
Again, great suggestions!